src/frontend/src/service/cookie-service.ts

```diff
@@ -0,0 +1,35 @@
+interface Cookie {
+  auth_token: string;
+  user_id: string;
+}
+
+export function decodeCookie() {
+  // this decodes the cookie creating an object with {key: `value`} pairs
+  // https://gist.github.com/rendro/525bbbf85e84fa9042c2#gistcomment-2784930
+  const data: any = document.cookie.split(';').reduce((res, c) => {
```

Comment:
Good point! I have refactored this a lot to make it a little more clear and return instead of save a value and return the value.

I have been trying to appease typescript and use some kind of whitelisting object but I have been unable to have it work with specific values expected as key names.

I have was able to simplify the typing in the `cookie-service.ts` file.

Reply:
Gotcha, Typescript stuff might be worth discussing at the meeting. I'd say just use `any` types if you get stuck on anything for more than a minute or two (I definitely have).